src/main/java/usdot/fhwa/stol/c2c/c2c_mvt/decoders/JsonDecoder.java

```diff
@@ -0,0 +1,182 @@
+package usdot.fhwa.stol.c2c.c2c_mvt.decoders;
+
+import com.github.erosb.jsonsKema.JsonParseException;
+import com.github.erosb.jsonsKema.JsonParser;
+import com.github.erosb.jsonsKema.JsonValue;
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import usdot.fhwa.stol.c2c.c2c_mvt.C2CMVTException;
+import usdot.fhwa.stol.c2c.c2c_mvt.controllers.StandardValidationController;
+import usdot.fhwa.stol.c2c.c2c_mvt.messages.JsonC2CMessage;
+
+/**
+ * Implementation of {@link Decoder} for Json Messages
+ * @author Aaron Cherney
+ */
+public class JsonDecoder extends Decoder
+{
+
+	/**
+	 * Decodes the message into a list of {@link JsonC2CMessage}s.
+	 * @param yBytes message(s) in bytes
+	 * @return A list of the decoded messages
+	 * @throws C2CMVTException
+	 */
+	@Override
+	public ArrayList<JsonC2CMessage> decode(byte[] yBytes)
+		throws C2CMVTException
+	{
+		if (!checkSecurity(yBytes))
+			throw new C2CMVTException(new Exception("Found possible security threat. Did not attempt validation."), null);
+		
+		ArrayList<byte[]> yMessages = separateMessages(yBytes);
+		
+		int nMsgCount = 0;
+		try
+		{
+			ArrayList<JsonC2CMessage> oMessages = new ArrayList(yMessages.size());
+			for (byte[] yMsg : yMessages)
+			{
+				oMessages.add(checkSyntax(yMsg));
+				++nMsgCount;
+			}
+			
+			// for testing purposing
+			{
+				for (JsonC2CMessage oMsg : oMessages)
+					StandardValidationController.addLogRecord(oMsg.getMessage());
+			} // remove this block before release
+			
+			return oMessages;
+		}
+		catch (JsonParseException oEx)
+		{
+			throw new C2CMVTException(oEx, String.format("Error occured while checking syntax of message %d of %d", nMsgCount + 1, yMessages.size()));
+		}
+	}
+
+	/**
+	 * Checks for multiple valid messages defined by the bytes. Messages can be separated
+	 * by an empty string, a single comma, or whitespace characters. Tests for
+	 * and ignores Byte order marking and ensures that the bytes could define
+	 * a JSON Object or Array by testing for matching {} or [] respectively
+	 * 
+	 * 
+	 * @param yBytes the message(s) in bytes
+	 * @return list of separate messages as byte[]
+	 * @throws C2CMVTException
+	 */
+	@Override
+	protected ArrayList<byte[]> separateMessages(byte[] yBytes)
+		throws C2CMVTException
+	{
+		ArrayList<byte[]> yMessages = new ArrayList();
+		try (ByteArrayInputStream oBaos = new ByteArrayInputStream(yBytes);
+			BufferedReader oReader = new BufferedReader(new InputStreamReader(oBaos, Charset.forName(m_sEncoding))))
```

Comment:
What kind of encoding we are using `m_sEncoding`? This `m_sEncoding` variable does not seem to be initialized.

Reply:
The encoding will be different for each standard. For JSON we are using UTF-8. The encoding gets set after the instantiation of the decoder with the Decoder#setEncoding() function